game/ui/general/savemenu.cpp

```diff
@@ -0,0 +1,304 @@
+			{
+				newItemSlot->Visible = false;
+			}
+		}
+	}
+	else
+	{
+		labelToHide = menuform->FindControlTyped<Label>("LABEL_LOADGAME");
+	}
+
+	if (labelToHide != nullptr)
+	{
+		labelToHide->Visible = false;
+	}
+
+	// load menu items
+	auto saves = saveManager.getSaveList();
+	auto existingSlotControl = saveListBox->FindControl(existingSaveItemId);
+	if (existingSlotControl != nullptr)
+	{
+		for (auto it = saves.begin(); it != saves.end(); ++it)
+		{
+			if (it->getType() != SaveType::Manual && currentAction == Save)
+			{ // skip quicksave and autosave on save screen
+				continue;
+			}
+
+			auto newControl = existingSlotControl->CopyTo(saveListBox);
+			newControl->SetData(mksp<SaveMetadata>(*it));
+			auto nameLabel = newControl->FindControlTyped<Label>("LABEL_NAME");
+			if (nameLabel != nullptr)
+			{
+				nameLabel->SetText(it->getName());
+			}
+
+			auto saveTimeLabel = newControl->FindControlTyped<Label>("LABEL_TIME");
+			if (saveTimeLabel != nullptr)
+			{
+				std::time_t timestamp = it->getCreationDate();
+				struct tm *tminfo = std::localtime(&timestamp);
+				std::stringstream ss;
+				if (timestamp != 0 && tminfo != nullptr)
+				{
+					ss << std::put_time(tminfo, "%c %Z");
+				}
+				saveTimeLabel->SetText(ss.str());
+			}
+
+			GameTime gameTime(it->getGameTicks());
+			auto gameDayLabel = newControl->FindControlTyped<Label>("LABEL_INGAME_DAY");
+			if (gameDayLabel != nullptr)
+			{
+				if (it->getGameTicks() == 0 && gameTime.getDay() == 0)
+				{
+					gameDayLabel->SetText("");
+				}
+				else
+				{
+					gameDayLabel->SetText("Day " + std::to_string(gameTime.getDay()));
+				}
+			}
+			auto gameTimeLabel = newControl->FindControlTyped<Label>("LABEL_INGAME_TIME");
+			if (gameTimeLabel != nullptr)
+			{
+				if (it->getGameTicks() == 0 && gameTime.getDay() == 0)
+				{
+					gameTimeLabel->SetText("");
+				}
+				else
+				{
+					auto seconds = it->getGameTicks() / TICKS_PER_SECOND;
+					auto minutes = seconds / 60;
+					auto hours = minutes / 60;
+
+					unsigned secondsClamped = seconds % 60;
+					unsigned minutesClamped = minutes % 60;
+					unsigned hoursClamped = hours % 24;
+
+					auto timeString = UString::format("%02u:%02u:%02u", hoursClamped,
+					                                  minutesClamped, secondsClamped);
+					gameTimeLabel->SetText(timeString);
+				}
+			}
+
+			auto difficultyLabel = newControl->FindControlTyped<Label>("LABEL_DIFFICULTY");
+			if (difficultyLabel != nullptr)
+			{
+				difficultyLabel->SetText(it->getDifficulty());
+			}
+
+			auto saveNameTextEdit = newControl->FindControlTyped<TextEdit>("TEXTEDIT_SAVE_NAME");
+			if (currentAction == Load && saveNameTextEdit != nullptr)
+			{
+				saveNameTextEdit->Visible = false;
+			}
+		}
+		existingSlotControl->Visible = false;
+	}
+}
+void SaveMenu::Pause() {}
+
+void SaveMenu::Resume() {}
+
+void SaveMenu::Finish() {}
+
+void ClearTextEdit(sp<TextEdit> textEdit)
+{
+	auto e = new OpenApoc::FormsEvent();
```

Comment:
This leaks 'e', you probably want to wrap this in a unique_ptr<> (up<> in openapoc namespace)